src/Maestro/DependencyUpdater/Program.cs

```diff
@@ -49,37 +49,29 @@ public static void Configure(IServiceCollection services)
                         .GetCustomAttribute<AssemblyInformationalVersionAttribute>()
                         ?.InformationalVersion);
             });
-            services.Configure<GitHubTokenProviderOptions>((options, provider) =>
+            services.Configure<GitHubTokenProviderOptions>("GitHub", (o, s) =>
             {
-                var config = provider.GetRequiredService<IConfiguration>();
-                IConfigurationSection section1 = config.GetSection("GitHub");
-                section1.Bind(options);
+                s.Bind(o);
             });
             services.AddGitHubTokenProvider();
 
             services.AddAzureDevOpsTokenProvider();
-            services.Configure<AzureDevOpsTokenProviderOptions>((options, provider) =>
```

Comment:
Is there anything we can do to prevent people from accidentally calling the "bad" one?

Reply:
I can't think of anything other than an analyzer. There is a possibility I could write something that would make calls to the bad one register as ambiguous, but that would be really hacky.